itests/fevm_test.go

```diff
@@ -119,22 +275,325 @@ func TestFEVMDelegateCall(t *testing.T) {
+	require.NoError(t, err)
+	_, _, err = client.EVM().InvokeContractByFuncName(ctx, fromAddr, contractAddr, "new_Test(string,uint256)", inputData)
+	t.Log(err)
+	require.NoError(t, err)
+
+	inputData, err = hex.DecodeString("0000000000000000000000000000000000000000000000000000000000000000")
+	require.NoError(t, err)
+
+	_, _, err = client.EVM().InvokeContractByFuncName(ctx, fromAddr, contractAddr, "get_Test_N(uint256)", inputData)
+	require.NoError(t, err)
+
+}
+
+// TestFEVMTestApp creates a contract that just has a self destruct feature and calls it
+func TestFEVMTestConstructor(t *testing.T) {
+	ctx, cancel, client := kit.SetupFEVMTest(t)
+	defer cancel()
+
+	//install contract Actor
+	filenameStorage := "contracts/Constructor.hex"
+	fromAddr, contractAddr := client.EVM().DeployContractFromFilename(ctx, filenameStorage)
+
+	//input = uint256{7}. set value and confirm tx success
+	inputData, err := hex.DecodeString("0000000000000000000000000000000000000000000000000000000000000007")
+	require.NoError(t, err)
+	_, _, err = client.EVM().InvokeContractByFuncName(ctx, fromAddr, contractAddr, "new_Test(uint256)", inputData)
+	t.Log(err)
+	require.NoError(t, err)
+
+}
+
+// TestFEVMAutoSelfDestruct creates a contract that just has a self destruct feature and calls it
+// XXX tx w destroy fails
+func TestFEVMAutoSelfDestruct(t *testing.T) {
+	ctx, cancel, client := kit.SetupFEVMTest(t)
+	defer cancel()
+
+	//install contract Actor
+	filenameStorage := "contracts/AutoSelfDestruct.hex"
+	fromAddr, contractAddr := client.EVM().DeployContractFromFilename(ctx, filenameStorage)
+
+	//call destroy
+	_, _, err := client.EVM().InvokeContractByFuncName(ctx, fromAddr, contractAddr, "destroy()", []byte{})
+	t.Log(err)
+	require.NoError(t, err) // XXX currently returns an error but should be success
+
+}
+
+// TestFEVMTestApp creates a contract that just has a self destruct feature and calls it
+// XXX calling self destruct fails
+func TestFEVMTestSendToContract(t *testing.T) {
+	ctx, cancel, client := kit.SetupFEVMTest(t)
+	defer cancel()
+
+	bal, err := client.WalletBalance(ctx, client.DefaultKey.Address)
+	require.NoError(t, err)
+
+	originalBalance, err := big.FromString("100000000000000000000000000")
+	require.NoError(t, err)
+	require.Equal(t, originalBalance, bal)
+
+	//install contract TestApp
+	filenameStorage := "contracts/SelfDestruct.hex"
+	fromAddr, contractAddr := client.EVM().DeployContractFromFilename(ctx, filenameStorage)
+
+	//transfer 1 wei to contract
+	sendAmount := big.NewInt(1)
+	sendMsg := &types.Message{
+		From:  fromAddr,
+		To:    contractAddr,
+		Value: sendAmount,
+	}
+	signedMsg, err := client.MpoolPushMessage(ctx, sendMsg, nil)
+	require.NoError(t, err)
+	mLookup, err := client.StateWaitMsg(ctx, signedMsg.Cid(), 3, api.LookbackNoLimit, true)
+	require.NoError(t, err)
+	require.Equal(t, exitcode.Ok, mLookup.Receipt.ExitCode)
```

Comment:
I don't think it's accurate to say we're transferring wei.

Also use the utility function you wrote here.

Reply:
thx! From your earlier comment I know they're attoFils! Also refactored into the util function

Reply:
ec750182b